plugins/playbackProfile.ts

```diff
@@ -0,0 +1,274 @@
+  const webmAudioCodecs = getSupportedWebMAudioCodecs(videoTestElement);
+
+  DirectPlayProfiles.push({
+    Container: 'webm',
+    Type: DlnaProfileType.Video,
+    AudioCodec: webmAudioCodecs.join(','),
+    VideoCodec: vpxVideoCodecs.join(',')
+  });
+
+  DirectPlayProfiles.push({
+    Container: 'mp4,m4v',
+    Type: DlnaProfileType.Video,
+    VideoCodec: mp4VideoCodecs.join(','),
+    AudioCodec: mp4AudioCodecs.join(',')
+  });
+
+  const supportedAudio = [
+    'opus',
+    'mp3',
+    'mp2',
+    'aac',
+    'flac',
+    'alac',
+    'webma',
+    'wma',
+    'wav',
+    'ogg',
+    'oga'
+  ];
+
+  for (const audioFormat of supportedAudio) {
+    if (audioFormat === 'mp2') {
+      DirectPlayProfiles.push({
+        Container: 'mp2,mp3',
+        Type: DlnaProfileType.Audio,
+        AudioCodec: audioFormat
+      });
+    } else if (audioFormat === 'mp3') {
+      DirectPlayProfiles.push({
+        Container: audioFormat,
+        Type: DlnaProfileType.Audio,
+        AudioCodec: audioFormat
+      });
+    } else {
+      DirectPlayProfiles.push({
+        Container: audioFormat === 'webma' ? 'webma,webm' : audioFormat,
+        Type: DlnaProfileType.Audio
+      });
+    }
+
+    // aac also appears in the m4a and m4b container
+    if (audioFormat === 'aac' || audioFormat === 'alac') {
+      DirectPlayProfiles.push({
+        Container: 'm4a,m4b',
+        AudioCodec: audioFormat,
+        Type: DlnaProfileType.Audio
+      });
+    }
+  }
+
+  return DirectPlayProfiles;
+}
+
+/**
+ * Returns a valid TranscodingProfile for the current platform.
+ *
+ * @returns {Array<TranscodingProfile>} An array of transcoding profiles for the current platform.
+ */
+function getTranscodingProfiles(): Array<TranscodingProfile> {
+  const TranscodingProfiles = [];
+
+  const hlsBreakOnNonKeyFrames = !!(
+    browserDetector.isApple() ||
+    browserDetector.isEdge() ||
+    !canPlayNativeHls(videoTestElement)
+  );
+
+  if (
+    canPlayNativeHls(videoTestElement) ||
+    browserDetector.supportsMediaSource()
+  ) {
+    TranscodingProfiles.push({
+      Container:
+        !canPlayNativeHls(videoTestElement) ||
+        browserDetector.isEdge() ||
+        browserDetector.isAndroid()
+          ? 'ts'
+          : 'aac',
+      Type: DlnaProfileType.Audio,
+      AudioCodec: 'aac',
+      Context: EncodingContext.Streaming,
+      Protocol: 'hls',
+      MaxAudioChannels: physicalAudioChannels.toString(),
+      MinSegments: browserDetector.isApple() ? 2 : 1,
+      BreakOnNonKeyFrames: hlsBreakOnNonKeyFrames
+    });
+  }
+
+  ['aac', 'mp3', 'opus', 'wav']
+    .filter(getSupportedAudioCodecs)
+    .forEach(function (audioFormat) {
+      TranscodingProfiles.push({
+        Container: audioFormat,
+        Type: DlnaProfileType.Audio,
+        AudioCodec: audioFormat,
+        Context: EncodingContext.Streaming,
+        Protocol: 'http',
+        MaxAudioChannels: physicalAudioChannels.toString()
+      });
+    });
```

Comment:
Should flac be included here? Or should this section if all formats are supported [see here](https://github.com/jellyfin/jellyfin-vue/pull/71/files#r487188754)

Reply:
This is a profile for formats we want to transcode to. We still support direct play for FLAC, but I'm not too sure what the point of transcoding to FLAC would be (it's also heavier for mobile users)

Reply:
Ah ok, that makes sense. Yeah transcoding from mp3 to flac would be pretty pointless